pipx/pipxrc.py

```diff
@@ -0,0 +1,193 @@
+import json
+import logging
+from pathlib import Path
+import re
+import textwrap
+from typing import List, Dict, NamedTuple, Any, Optional
+
+from pipx.Venv import PipxVenvMetadata, Venv
+from pipx.util import PipxError
+
+
+PIPX_INFO_FILENAME = "pipxrc.json"
+
+
+class JsonEncoderHandlesPath(json.JSONEncoder):
+    def default(self, obj):
+        # only handles what json.JSONEncoder doesn't understand by default
+        if isinstance(obj, Path):
+            return {"__type__": "Path", "__Path__": str(obj)}
+        return super().default(obj)
+
+
+def _json_decoder_object_hook(json_dict):
+    if json_dict.get("__type__", None) == "Path" and "__Path__" in json_dict:
+        return Path(json_dict["__Path__"])
+    return json_dict
+
+
+class PackageInfo(NamedTuple):
+    package_or_url: Optional[str]
+    pip_args: List[str]
+    include_dependencies: bool
+    include_apps: bool
+
+
+class Pipxrc:
+    def __init__(self, venv_dir: Path, read: bool = True):
+        self.venv_dir = venv_dir
+        self.reset()
+        if read:
+            self.read()
+
+    def reset(self) -> None:
+        # We init this instance with reasonable fallback defaults for all
+        #   members, EXCEPT for those we cannot know:
+        #       self.main_package.package_or_url=None
+        #       self.venv_metadata.package_or_url=None
+        self.main_package = PackageInfo(
+            package_or_url=None,
+            pip_args=[],
+            include_dependencies=False,
+            include_apps=True,  # always True for main_package
+        )
+        self.venv_metadata: Optional[PipxVenvMetadata] = None
+        self.venv_args: List[str] = []
+        self.injected_packages: List[PackageInfo] = []
+        # Only change this if file format changes
+        self._pipxrc_version: str = "0.1"
```

Comment:
all of the member variables should be declared in __init__

Reply:
Done.

Question: I'm always torn between best practice of putting the init of all variables in the `__init__()` and having only one set of code that does the reset (i.e. `reset()`).  Is there a better way of handling this, or is just duplicating the member initialization code the way?